crates/primitives/src/transaction/mod.rs

```diff
@@ -897,9 +897,18 @@ mod tests {
     };
     use bytes::BytesMut;
     use ethers_core::utils::hex;
-    use reth_rlp::{Decodable, Encodable};
+    use reth_rlp::{Decodable, DecodeError, Encodable};
     use std::str::FromStr;
 
+    #[test]
+    fn test_decode_empty_typed_tx() {
+        let input = [0x80u8];
+        let res = TransactionSigned::decode(&mut &input[..]).unwrap_err();
+        // TODO consider `let expected = DecodeError::InputTooShort;`
```

Comment:
Open to it, either in this or separate PR

Reply:
I'll revise it in this PR.

Reply:
Done w/ b8e54d30fb38c5c3ce317a49da2ad56bf25cb173